lib/theme_check/remote_asset_file.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+require "net/http"
+require "pathname"
+require "zlib"
+
+module ThemeCheck
+  class RemoteAssetFile
+    class << self
+      def cache
+        @cache ||= {}
+      end
+
+      def from_src(src)
+        key = uri(src).to_s
+        cache[key] = RemoteAssetFile.new(src) unless cache.key?(key)
+        cache[key]
+      end
+
+      def uri(src)
+        URI.parse(src.sub(%r{^//}, "https://"))
+      end
+    end
+
+    def initialize(src)
+      @uri = RemoteAssetFile.uri(src)
+      @content = nil
+    end
+
+    def content
+      return @content unless @content.nil?
+
+      res = Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == 'https') do |http|
+        req = Net::HTTP::Get.new(@uri)
+        req['Accept-Encoding'] = 'gzip, deflate, br'
+        http.request(req)
+      end
+
+      @content = res.body
+    end
+
+    def gzipped_size
+      @gzipped_size ||= @content.size
```

Comment:
`bytesize` would be more accurate, especially for encoded content: https://ruby-doc.org/core-3.0.0/String.html#method-i-bytesize